spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/ops/Floor.scala

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2016 The BigDL Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.intel.analytics.bigdl.nn.ops
+
+import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric
+import com.intel.analytics.bigdl.tensor._
+
+import scala.reflect.ClassTag
+
+class Floor[T: ClassTag]()
+  (implicit ev: TensorNumeric[T]) extends Operation[Tensor[T], T] {
```

Comment:
Should we change the Operation signature so that the output type is not coupled with type parameter T. CC @yiheng